editor/src/clj/editor/workspace.clj

```diff
@@ -227,6 +214,25 @@ ordinary paths."
       (when-let [workspace (:workspace base-resource)]
         (resolve-workspace-resource workspace path)))))
 
+(defn- template-path [resource-type]
+  (or (:template resource-type)
+      (some->> resource-type :ext (str "templates/template."))))
+
+(defn get-template [workspace resource-type]
+  (let [path (template-path resource-type)
+        java-resource (when path (io/resource path))
+        editor-resource (when path (find-resource workspace path))]
+    (or java-resource editor-resource)))
```

Comment:
I imagine I'll do something similar for the icons: find it either in the java package or the project resources